### PR TITLE
add back page counts

### PR DESCRIPTION
**Pull request by zsblevins**
## Description

Add back paging info leveraging Temporal's count query for more efficient pagination data than the old implementation

## Validation

image

- [x] Standard CI passes.
- [x] Kind integration passes, or this PR explains why it was not run.

The kind integration test is manual due to taking ~30 min to complete. When the PR is ready for review,
run Actions -> Kind Integration -> Run workflow against the copy-pr-bot generated
`pull-request/<PR_NUMBER>` branch. Use the default `test_path` for the full suite,
or narrow it only while debugging.

Passing Kind Integration run:
https://github.com/NVIDIA/nv-config-manager/actions/runs/28120975254/job/83272545771

## Checklist

- [x] I am familiar with the contributing guidelines in `CONTRIBUTING.md`.
- [x] Commits are signed off for DCO compliance.
- [x] New or existing tests cover these changes, or the PR explains why tests are not needed.
- [x] Documentation is updated for user-facing behavior changes.
- [x] Generated artifacts are updated when applicable, such as OpenAPI specs,
      docs screenshots, or Helm/rendered outputs.

## Summary by CodeRabbit

* **New Features**
  * Workflow list responses now return pagination metadata (`total_count`, `page_count`) and the UI footer updates accordingly.
  * Added an optional “hide completed workflows” filter to the workflow listing.
* **Bug Fixes**
  * End-time filtering is now based on each workflow’s close time (with handling for missing/invalid close times).
  * The workflow listing now returns HTTP 400 when `limit` is non-positive.
* **Tests**
  * Updated/extended unit, UI mock, and E2E coverage for pagination metadata, hide-completed behavior, and close-time filtering edge cases.

### CODE REVIEW

File: docs/api-specs/temporal.openapi.json [hunk 1]
```diff
@@ -1891,7 +1899,9 @@
         },
         "required": [
           "workflows",
-          "next_page_token"
+          "next_page_token",
+          "total_count",
+          "page_count"
         ],
```
Comment (ryanheffernan) [inline docs/api-specs/temporal.openapi.json:1905]: looks like this needs a regen to pick up hide_completed

File: ui/tests/docs-screenshots/workflowForms.spec.ts
Comment (ryanheffernan) [inline ui/tests/docs-screenshots/workflowForms.spec.ts]: I think these need to be updated for the new CloseTime behavior?

**Review by ryanheffernan — DISMISSED** [review]

**Review by ryanheffernan — APPROVED** [review]
